src/resultset.rs

```diff
@@ -8,6 +8,27 @@ use value::ToMysqlValue;
 use writers;
 use {Column, ErrorKind, StatementData};
 
+/// Convenience type for responding to a client `USE <db>` command.
+pub struct InitWriter<'a, W: Write + 'a> {
+    pub(crate) writer: &'a mut PacketWriter<W>,
+}
+
+impl<'a, W: Write + 'a> InitWriter<'a, W> {
+    /// Tell client that database context has been changed
+    pub fn ok(self) -> io::Result<()>
+    {
+        writers::write_ok_packet(self.writer, 0, 0, StatusFlags::empty())
+    }
+
+    /// Tell client that there was a problem changing the database context.
+    pub fn error<E>(self, kind: ErrorKind, msg: &E) -> io::Result<()>
+    where
+        E: Borrow<[u8]> + ?Sized,
+    {
+        writers::write_err(kind, msg.borrow(), self.writer)
```

Comment:
I wonder if we want to require that the `kind` is `ErrorKind::ER_BAD_DB_ERROR` here? Does it make sense to want to return other error kinds in this context?

Reply:
I did have it forced to ER_BAD_DB_ERROR but when I re-read the spec it just said returns an Ok_packet or ERR_packet. I did a brief skim of the MySQL server source and found that it can also potentially return ER_DBACCESS_DENIED. 

It ends up looking like this for the implementation:

    fn on_init(&mut self, schema: &str, writer: InitWriter<W>) -> io::Result<()> {
        if schema == "foobar" {
            writer.ok()
        }else{
            writer.error(ErrorKind::ER_BAD_DB_ERROR, "So sad, so bad".as_bytes())
        }
    }

That having been said: I don't have a strong preference either way. I can imagine a scenario where someone might want to return any arbitrary error code, and I can also imagine the scenario where they probably shouldn't.

Reply:
I suspect the best resolution here is to include information about what kind the user _probably_ wants to use in the documentation for `InitWriter::error`, and that should be good enough :)